core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/DistanceRangeMap.kt

```diff
@@ -35,6 +36,15 @@ interface DistanceRangeMap<T> : Iterable<DistanceRangeMap.RangeMapEntry<T>> {
     /** Get the value at the given offset, if there is any.
      * On exact transition offsets, the value for the higher offset is used. */
     fun get(offset: Distance): T?
+
+    /** Returns a new DistanceRangeMap of the ranges between lower and upper */
+    fun subMap(lower: Distance, upper: Distance): DistanceRangeMap<T>
+
+    /** Returns the first map updated with another, using a merge function to fuse the values of intersecting ranges*/
+    fun <U> updateMap(
```

Comment:
When reading the method name and docstring, I'd assume this would update the given map, but it creates a new one. 

Looking at how it's used, I think it should update the map instead of a copy. We could also add a `clone` method for the cases where we actually want a copy.